server/datastore/mysql/software_installers.go

```diff
@@ -2316,3 +2316,42 @@ WHERE
 
 	return res, nil
 }
+
+func (ds *Datastore) GetSoftwareInstallerByHash(ctx context.Context, sha256 string) (map[uint]*fleet.ExistingSoftwareInstaller, error) {
+	stmt := `
+SELECT 
+	si.id AS installer_id,
+	si.team_id AS team_id,
+	si.filename AS filename,
+	si.extension AS extension,
+	si.version AS version,
+	si.platform AS platform,
+	st.source AS source,
+	st.bundle_identifier AS bundle_identifier,
+	st.name AS title
+FROM
+	software_installers si
+	JOIN software_titles st ON si.title_id = st.id
+WHERE
+	storage_id = ?` // TODO(JVE): do we need to include URL in this search? If so, do we need to change the index?
```

Comment:
Yep; if the URL is supplied, the URL should be included.

Don't think we need any index changes here, as the SHA will filter the result set down enough on its own. Also, URLs are long enough that they would break an index if we tried to use them, whereas a 64-character hash is NBD.